### ISSUE TITLE
Swap test

### ISSUE THREAD

**Issue by andrewmusselman** [apache/mahout#477]

**Comment by shashikant-op** [comment]
if  this issue is still open then can i work on it

**Comment by rawkintrevo** [comment]
Yes, I had it assigned to me to remember to do something with it, and then got distracted with FOSDEM (then some personal stuff immediately following that). I've done nothing of note on it. @shashikant-op if you want to take it go ahead.

**Comment by andrewmusselman** [comment]
@shashikant-op go for it, thanks for volunteering!
